### PR TITLE
Alasql 1030 - Make sure errors bubble up to promiseExec

### PR DESCRIPTION
**Pull request by elipalmer**
Corrects issue found in #1030.  I found that the error callback function in loadFile was not being passed in or used.  The exception thrown was being unhandled causing the process to die and the error being sent to the console.

I made a note in the code but will mention it here as well.  The promiseExec function has the (err, data) node callback format standard swapped.  I tried switching them, but that resulted in non errors being sent as errors from elsewhere in the code.  I don't yet have a strong enough understanding of the code base to be comfortable switching them and knowing what else needs to be corrected.

There were a lot of spacing inconsistencies so I ran auto indent to normalize everything in the files I touched.  I also ran eslint --fix on the files I touched.

Running the unit tests against master resulted in this:
 1452 passing (10s)
  376 pending

Running the unit tests against the branch resulted in this:
 1456 passing (9s)
  376 pending

It's not clear to me what the pending tests are, but It seems I didn't break any existing tests.

I did not include the dist/ files as it seems those are generated.

### CODE REVIEW

**Comment by elipalmer** [comment]
OK, got CI tests to pass.  I didn't realize older versions of node needed to be supported.  Removed async/await references, this should be ready for review.

Thanks!

File: src/15utility.js [hunk 1]
```diff
@@ -818,29 +822,29 @@ function isIE() {
...
+ @return {integer} hash number
+ */
 
 // FNV-1a inspired hashing
 var hash = (utils.hash = function(str) {
 	var hash = 0x811c9dc5,
 		i = str.length;
 	while (i) {
-		hash = hash ^ str.charCodeAt(--i);
+		hash ^= str.charCodeAt(--i);
```
Comment (mathiasrw) [inline src/15utility.js]: Neat

File: src/40select.js [hunk 2]
```diff
@@ -368,9 +368,13 @@ yy.Select.prototype.compile = function(databaseid, params) {
 	// Now, compile all togeather into one function with query object in scope
 	var statement = function(params, cb, oldscope) {
 		query.params = params;
-		var res1 = queryfn(query, oldscope, function(res) {
-			//console.log(res[0].schoolid);
-			//console.log(184,res);
+		// Note the callback function has the data and error reversed due to existing code in promiseExec which has the
```
Comment (mathiasrw) [inline src/40select.js]: Great with a good comment. I made the promiseExec myself - so I'm glad you found this bug!

File: src/84from.js [hunk 3]
```diff
@@ -350,7 +350,7 @@ if(false) {
 		parseText(contents);
 	} else {
 		contents = alasql.utils.autoExtFilename(contents, 'csv', opts);
-		alasql.utils.loadFile(contents, !!cb, parseText);
+		alasql.utils.loadFile(contents, !!cb, parseText, query.cb);
```
Comment (mathiasrw) [inline src/84from.js]: Neat.

File: test/test624CSVParsing.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+if (typeof exports === 'object') {
```
Comment (mathiasrw) [inline test/test624CSVParsing.js]: Thank you for making a good test file too.

**Comment by elipalmer** [comment]
Awesome, thanks for merging!  Glad you had a successful "no computer" vacation, I haven't had one of those in years.

**Pull request merged**